src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,704 @@
+        dfcx_pages_map = {}
+        for flow_id in dfcx_flows_map.keys():
+            dfcx_pages_map[flow_id] = dfcx_pages.get_pages_map(flow_id=flow_id)
+        commons_config["test_cases"] = dfcx_testcases
+        commons_config["flows"] = dfcx_flows
+        commons_config["pages"] = dfcx_pages
+        commons_config["intents"] = dfcx_intents
+        commons_config["flows_map"] = dfcx_flows_map
+        commons_config["pages_map"] = dfcx_pages_map
+        commons_config["intents_map"] = dfcx_intents_map
+        commons_config["flows_map_reverse"] = (
+            self._reverse_dict(dict_=commons_config["flows_map"]))
+        commons_config["intents_map_reverse"] = (
+            self._reverse_dict(dict_=commons_config["intents_map"]))
+        commons_config["pages_map_reverse"] = (
+            {f_id:self._reverse_dict(dict_=pages_map) \
+             for f_id, pages_map in dfcx_pages_map.items()})
+
+        return commons_config
+
+    def _get_new_test_config(
+        self,
+        source: dict,
+        target: dict,
+        test_case: types.TestCase) -> Union[types.TestConfig, None]:
+        """Update the test case types.test_config to the new types.test_config
+            Update the uuids of flow, page, intent by the display_name
+          args:
+            source: a dictionary of the source agent's flows, pages, and intents
+            target: a dictionary of the target agent's flows, pages, and intents
+            test_case: types.test_config object
+          returns:
+            types.test_config
+        """
+        source_flow_id = self._get_flow_id_from_test_config(test_case)
+        source_page_id = self._get_page_id_from_test_config(
+            test_case, source_flow_id)
+        source_flow = self._get_flow_display_name_by_id(
+            source_flow_id, source["flows_map"])
+        source_page = self._get_page_display_name_by_id(
+            source_flow_id, source_page_id, source["pages_map"])
+        target_test_config = types.TestConfig()
+
+        if test_case.test_config.flow:
+            target_flow_id = self._get_flow_id_by_flow_name(
+                flow_name=source_flow,
+                flows_map=target["flows_map_reverse"])
+            if target_flow_id:
+                target_test_config.flow = target_flow_id
+            else:
+                return None
+
+        elif test_case.test_config.page:
+            target_flow_id = self._get_flow_id_by_flow_name(
+                flow_name=source_flow,
+                flows_map=target["flows_map_reverse"])
+            target_page_id = self._get_page_id_by_page_name(
+                flow_id=target_flow_id,
+                page_name=source_page,
+                pages_map=target["pages_map_reverse"])
+            if target_page_id:
+                target_test_config.page = target_page_id
+            else:
+                return None
+
+        return target_test_config
+
+    def _get_flow_id_from_test_config(
```

Comment:
@jkshj21 this is redundant code that is [already implemented in the TestCases ](https://github.com/GoogleCloudPlatform/dfcx-scrapi/blob/main/src/dfcx_scrapi/core/test_cases.py#L81) class.

Instead of reimplementing here, just access it from `self._global_test_cases` instead.

Reply:
Removed and it uses the dfcx_scrapi.core.test_cases class to call this function